scylla/src/transport/session.rs

```diff
@@ -280,6 +280,25 @@ impl Session {
         query: impl Into<Query>,
         values: impl ValueList,
     ) -> Result<Option<Vec<result::Row>>, QueryError> {
+        let query: Query = query.into();
+        let query_text: &str = query.get_contents();
+
+        // In case the user tried doing session.query("use keyspace ks") run session::use_keyspace
+        if query_text.to_lowercase().starts_with("use ") {
+            eprintln!("Warning: Raw USE KEYSPACE queries are experimental, please use session::use_keyspace instead");
+
+            let mut keyspace_name: &str = &query_text["use ".len()..].trim_end_matches(';').trim();
+
+            let case_sensitive: bool = keyspace_name.starts_with('"');
+
+            keyspace_name = keyspace_name.trim_matches('"');
```

Comment:
I wonder if, instead of parsing USE statements, we could do the following:

- Run the statement verbatim,
- Detect that we got [`Set_keyspace`](https://github.com/apache/cassandra/blob/dfdb6e3d8ebd797bbda8bbe5e47b5228a9281839/doc/native_protocol_v4.spec#L652) response,
- Run `use_keyspace()` with the name we got in the response.

WDUT @cvybhu @psarna ? This has worse latency because we have to do two round-trips, but does not need any parsing on our side.

Reply:
I'm also fine if we do it in a separate issue.

Reply:
I agree, if we manage to not parse anything driver-side, it would be very nice. The CQL protocol does not make it particularly easy, but we should take the opportunity if possible - and here we have a specific response to handle. And I also agree it can be done as a follow-up.

Reply:
This sounds like a good idea, I'm still not sure if this parsing is 100% correct.
I will create a follow-up issue for this.

Reply:
I created #168